src/PrestaShopBundle/Form/Admin/Type/DateRangeLocalType.php

```diff
@@ -0,0 +1,151 @@
+     */
+    protected $translator;
+
+    /**
+     * @var FormCloner
+     */
+    protected $formCloner;
+
+    /**
+     * @var string
+     */
+    protected $dateFormatFull;
+
+    /**
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        FormCloner $formCloner,
+        string $dateFormatFull
+    ) {
+        $this->translator = $translator;
+        $this->formCloner = $formCloner;
+        $this->dateFormatFull = $dateFormatFull;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $now = new DateTime();
+        $builder
+            ->add('from', DatePickerType::class, [
+                'required' => false,
+                'label' => $this->translator->trans('Start date', [], 'Admin.Global'),
+                'attr' => [
+                    'placeholder' => $options['attr']['placeholder'] ? $options['attr']['placeholder'] : $this->translator->trans('YY-MM-DD', [], 'Admin.Global'),
+                    'class' => 'from date-range-start-date',
+                    'data-default-value' => $now->format($this->dateFormatFull),
+                ],
+                'date_format' => $options['date_format'],
+            ])
+            ->add('to', DatePickerType::class, [
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $options['attr']['placeholder'] ? $options['attr']['placeholder'] : $this->translator->trans('YY-MM-DD', [], 'Admin.Global'),
+                    'class' => 'to date-range-end-date',
+                    'data-default-value' => $now->format($this->dateFormatFull),
+                ],
+                'label' => $this->translator->trans('End date', [], 'Admin.Global'),
+                'date_format' => $options['date_format'],
+            ])
+        ;
+
+        if ($options['has_unlimited_checkbox']) {
+            $builder->add('unlimited', CheckboxType::class, [
+                'label' => $this->translator->trans('Unlimited', [], 'Admin.Global'),
+                'required' => false,
+                'attr' => [
+                    'class' => 'date-range-unlimited',
+                ],
+            ]);
+
+            $builder->addEventListener(FormEvents::PRE_SET_DATA, [$this, 'adaptUnlimited']);
+            $builder->addEventListener(FormEvents::PRE_SUBMIT, [$this, 'adaptUnlimited']);
+        }
+
+        $builder->get('from')->addModelTransformer(new CallbackTransformer(
+            function ($from) {
+                if ($from !== null && $from != '0000-00-00 00:00:00') {
+                    $dateTime = DateTime::createFromFormat('Y-m-d h:i:s', $from);
+
+                    return $dateTime->format($this->dateFormatFull);
+                }
+
+                return $from;
+            },
+            function ($from) {
+                if ($from !== null) {
```

Comment:
In this case it can be '0000-00-00 00:00:00'?
Because if it is, then the $dateTime->format() will probably return invalid value.